src/OpenLoco/Windows/CompanyWindow.cpp

```diff
@@ -2278,12 +2287,95 @@ namespace OpenLoco::Ui::Windows::CompanyWindow
             Common::repositionTabs(self);
         }
 
+        // 0x00437D60
+        // Probably converts performance index * 10 to corporate rating index
+        // for example 10.0% (performanceIndex=100) converts to rating "Engineer" (output=1)
+        // input:
+        // ax = performanceIndex
+        // output:
+        // ax
+        static int16_t sub_437D60(int16_t performanceIndex)
+        {
+            registers regs;
+            regs.ax = performanceIndex;
+            call(0x00437D60, regs);
+            return regs.ax;
+        }
+
         // this will prepare _commonFormatArgs array before drawing the StringIds::challenge_value
         // after that for example it will draw this string: Achieve a performance index of 10.0% ("Engineer")
+        // Note: no input and output parameters are in the original assembly, update is done in the memory
+        // in this implementation we return FormatArguments so in the future it will be not depending on global variables
         // 0x004384E9
-        static void sub_4384E9()
+        static FormatArguments sub_4384E9()
         {
-            call(0x004384E9);
+            FormatArguments args{};
+
+            // 004384ED
+            switch (objectiveType)
+            {
+                case 0: // 004384FB-0043850A
+                    args.push(StringIds::achieve_a_company_value_of);
+                    args.push(*objectiveCompanyValue);
+                    break;
+
+                case 1: // 0043850C-0043851B
+                    args.push(StringIds::achieve_a_monthly_profit_from_vehicles_of);
+                    args.push(*objectiveMonthlyVehicleProfit);
+                    break;
+
+                case 2: // 0043851D-00438543
+                {
+                    args.push(StringIds::achieve_a_performance_index_of);
+                    int16_t performanceIndex = objectivePerformanceIndex * 10;
+                    args.push(performanceIndex);
+                    string_id corporateRating = sub_437D60(performanceIndex) + StringIds::corporate_rating_platelayer;
+                    args.push(corporateRating);
+                    break;
+                }
+
+                case 3: //00438545-00438576
+                {
+                    args.push(StringIds::deliver);
+                    cargo_object* cargoObject = _50D15C;
+                    if (objectiveDeliveredCargoType != 0xFF)
+                    {
+                        cargoObject = _cargoObjects[objectiveDeliveredCargoType];
+                    }
+                    args.push(cargoObject->unit_name_plural);
+                    args.push(*objectiveDeliveredCargoAmount);
+                    break;
+                }
+            }
+
+            if ((objectiveFlags & Scenario::objective_flags::flag_0) != 0) // 0043857B
```

Comment:
Could you shuffle all of the names down so that it is as follows:
```
be_top_company = (1 << 0)
be_within_top_three_companies = (1<< 1)
within_time_limit = (1 << 2)
```

Reply:
Please check after this update if the scenario options windows is working correctly.